modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/util/GroupUtil.java

```diff
@@ -32,25 +32,31 @@ public static Long getDepotGroupId(
 		DepotEntryLocalService depotEntryLocalService,
 		GroupLocalService groupLocalService) {
 
-		Group group = groupLocalService.fetchGroup(companyId, assetLibraryKey);
+		Group group = groupLocalService.fetchGroup(
+			Long.valueOf(assetLibraryKey));
 
 		if (group == null) {
-			try {
-				DepotEntry depotEntry = depotEntryLocalService.fetchDepotEntry(
-					GetterUtil.getLong(assetLibraryKey));
+			group = groupLocalService.fetchGroup(companyId, assetLibraryKey);
 
-				if (depotEntry == null) {
-					return null;
-				}
+			if (group == null) {
+				try {
+					DepotEntry depotEntry =
+						depotEntryLocalService.fetchDepotEntry(
+							GetterUtil.getLong(assetLibraryKey));
 
-				group = depotEntry.getGroup();
-			}
-			catch (PortalException portalException) {
-				if (_log.isDebugEnabled()) {
-					_log.debug(portalException);
+					if (depotEntry == null) {
+						return null;
+					}
+
+					group = depotEntry.getGroup();
 				}
+				catch (PortalException portalException) {
+					if (_log.isDebugEnabled()) {
+						_log.debug(portalException);
+					}
 
-				return null;
+					return null;
+				}
```

Comment:
There's too much nesting here now. Can you move the logic into a `_getGroup` method and use early returns?
```
Group = _getGroup(...);

if (_checkGroup(group)) {
    return group.getGroupId();
}

return null;

...
private Group _getGroup(...) {
    Group group = groupLocalService.fetchGroup(companyId, assetLibraryKey);

    if (group != null) {
        return group;
    }

    group = groupLocalService.fetchGroup(companyId, assetLibraryKey);

    if (group != null) {
        return group;
    }

   ...
```
}